frontend/packages/kubevirt-plugin/src/k8s/enhancedK8sMethods/__tests__/enhancedK8sMethods.spec.ts

```diff
@@ -0,0 +1,205 @@
+import { EnhancedK8sMethods } from '../enhancedK8sMethods';
+import { HistoryItem, HistoryType } from '../types';
+
+import { VirtualMachineModel } from '../../../models';
+import { VMWrapper } from '../../wrapper/vm/vm-wrapper';
+import { PatchBuilder } from '@console/shared/src/k8s';
+import { K8sKind, K8sResourceCommon, Patch } from '@console/internal/module/k8s';
+import { K8sKillError } from '../errors';
+
+const disableHistoryOpts = { disableHistory: true };
+
+const spyEnhancedK8sMethods = () => {
+  const methods = new EnhancedK8sMethods();
+  // mocks the approximate behaviour of CRUD methods
+  const methodsUnchecked = methods as any;
+  methodsUnchecked.k8sCreate = async (
+    kind: K8sKind,
+    data: K8sResourceCommon,
+    opts,
+    enhancedOpts,
+  ) => {
+    methodsUnchecked.registerKind(kind);
+    methodsUnchecked.appendHistory(new HistoryItem(HistoryType.CREATE, data), enhancedOpts);
+  };
+  // different behaviour than the original - we need to store the testVM
+  methodsUnchecked.k8sGet = async (kind: K8sKind, data: K8sResourceCommon, opts, enhancedOpts) => {
+    methodsUnchecked.registerKind(kind);
+    methodsUnchecked.appendHistory(new HistoryItem(HistoryType.GET, data), enhancedOpts);
+  };
+  methodsUnchecked.k8sPatch = async (
+    kind: K8sKind,
+    data: K8sResourceCommon,
+    patches: Patch[],
+    enhancedOpts,
+  ) => {
+    methodsUnchecked.registerKind(kind);
+    methodsUnchecked.appendHistory(
+      new HistoryItem(HistoryType.PATCH, { ...data, ...patches[0] }),
+      enhancedOpts,
+    );
+  };
+  methodsUnchecked.k8sKill = async (
+    kind: K8sKind,
+    data: K8sResourceCommon,
+    opts,
+    json,
+    enhancedOpts,
+  ) => {
+    methodsUnchecked.registerKind(kind);
+    methodsUnchecked.appendHistory(new HistoryItem(HistoryType.DELETE, data), enhancedOpts);
+  };
+  return { methods, methodsUnchecked };
```

Comment:
made a different method in the spyEnhancedK8sMethods
methods.mockK8sGet to avoid the type incompatibilities 

using just methods outside of this one and using any typecasts